pkg/provisioner/ironic/validatemanagementaccess_test.go

```diff
@@ -95,3 +96,79 @@ func TestValidateManagementAccessCreateNode(t *testing.T) {
 	assert.NotEqual(t, "", host.Status.Provisioning.ID)
 	assert.Equal(t, ironic.CreatedNodes[0].UUID, host.Status.Provisioning.ID)
 }
+
+func TestValidateManagementAccessLinkExistingIronicNodeByMAC(t *testing.T) {
+	// Create an Ironic node, and then create a host with a matching MAC
+	// Test to see if the node was found, and if the link is made
+
+	existingNode := nodes.Node{
+		UUID: "33ce8659-7400-4c68-9535-d10766f07a58",
+	}
+
+	existingNodePort := ports.Port{
+		NodeUUID: existingNode.UUID,
+		Address:  "11:11:11:11:11:11",
+	}
+
+	ironic := testserver.NewIronic(t).Ready().CreateNodes().WithNode(existingNode).Port(existingNodePort)
+	ironic.Start()
+	defer ironic.Stop()
+
+	host := makeHost()
+	host.Spec.BootMACAddress = "11:11:11:11:11:11"
+	host.Status.Provisioning.ID = "" // so we don't lookup by uuid
+
+	auth := clients.AuthConfig{Type: clients.NoAuth}
+	prov, err := newProvisionerWithSettings(host, bmc.Credentials{}, nullEventPublisher,
+		ironic.Endpoint(), auth, testserver.NewInspector(t).Endpoint(), auth,
+	)
+	if err != nil {
+		t.Fatalf("could not create provisioner: %s", err)
+	}
+
+	result, err := prov.ValidateManagementAccess(false)
+	if err != nil {
+		t.Fatalf("error from ValidateManagementAccess: %s", err)
+	}
+	assert.Equal(t, "", result.ErrorMessage)
+	assert.NotEqual(t, "", host.Status.Provisioning.ID)
+
+	// No new nodes should have been created
+	assert.Equal(t, len(ironic.CreatedNodes), 0)
+}
+
+func TestValidateManagementAccessExistingPortWithWrongUUID(t *testing.T) {
```

Comment:
Sorry, I left a confused comment last time. What we want is to test cases where there is a port, but the port doesn't match what we're using to look it up. I said the UUID, but I should have said MAC.

The case you have here is another good one, but I would expect to have Ironic report an error because there is already a port using the MAC but that is not associated with our node. Therefore when we try to create a new node, it would fail because a port with that MAC already exists.

Reply:
It looks like the `findExistingNode` method has some overloaded semantics that need to be fixed.

Reply:
The test calls `ValidateManagementAccess()`, which does both the lookup and the create. I suspect it's working because our mock server implementation is incomplete so it's not reporting the error that ironic would.

Reply:
Oh, duh, and now I see that on 170 you're calling `assert.Error()` so the test *is* expecting an error. Do we know it's the right error?

Reply:
I added an explicit check for the error kind.